src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -458,7 +458,10 @@ private Stream<SourceFile> processTestSources(
         javaParserBuilder.classpath(testDependencies).typeCache(typeCache);
         kotlinParserBuilder.classpath(testDependencies).typeCache(new JavaTypeCache());
 
-        List<Path> testJavaSources = listJavaSources(mavenProject.getBasedir().toPath().resolve(mavenProject.getBuild().getTestSourceDirectory()));
+        Set<Path> testJavaSources = new LinkedHashSet<>(listJavaSources(mavenProject.getBasedir().toPath().resolve(mavenProject.getBuild().getTestSourceDirectory())));
```

Comment:
Why wrap it in a set? :)

Reply:
It is testJavaSource used later in the function, after this change it can contain duplicates, which will break things.

Reply:
No longer needed after [clearing out duplicates](https://github.com/openrewrite/rewrite-maven-plugin/pull/888/commits/093f3456e72ef33fceb7f904b1262b338c88b530).